kornia/contrib/connected_components.py

```diff
@@ -0,0 +1,44 @@
+import torch
+import torch.nn.functional as F
+
+
+def connected_components(input: torch.Tensor, num_iterations: int = 100) -> torch.Tensor:
+    r"""Computes the Connected-component labelling (CCL) algorithm.
+
+    The implementation is an adaptation of the following repository:
+
+    https://gist.github.com/efirdc/5d8bd66859e574c683a504a4690ae8bc
+
+    Args:
+        input: the binarized input image with shape :math:`(B, 1, H, W)`.
```

Comment:
How about using the decorator `perform_keep_shape_image` to support any image dimension?